components/trade/AdvancedTradeForm.tsx

```diff
@@ -628,6 +629,42 @@ const AdvancedTradeForm = () => {
       tickDecimals,
     ],
   )
+  const calcOrderPrice = useCallback(
+    (price: number, orderbook: OrderbookL2) => {
+      let orderPrice = price
+      if (tradeForm.tradeType === 'Market') {
+        try {
+          if (tradeForm.side === 'sell') {
+            const marketPrice = Math.max(
+              oraclePrice,
+              orderbook?.bids?.[0]?.[0] || 0,
+            )
+            orderPrice = marketPrice * (1 - MAX_PERP_SLIPPAGE)
+          } else {
+            const marketPrice = Math.min(
+              oraclePrice,
+              orderbook?.asks?.[0]?.[0] || Infinity,
+            )
+            orderPrice = marketPrice * (1 + MAX_PERP_SLIPPAGE)
+          }
+        } catch (e) {
```

Comment:
hmm not sure if we should use 0 and infinity, if they were to be used then it would mean full loss of funds in that token or market, no?